content/common/projects/grade.sh

```diff
@@ -0,0 +1,136 @@
+#!/bin/bash
+# SPDX-License-Identifier: BSD-3-Clause
+
+# Grade style based on build warnings and linter warnings / errors.
+# Points are subtracted from the maximum amount of style points (10).
+#   - For 15 or more build warnings, all points (10) are subtracted.
+#   - For [10,15) build warnings, 6 points are subtracted.
+#   - For [5,10) build warnings, 4 points are subtracted.
+#   - For [1,5) build warnings, 2 points are subtracted.
+#   - For 25 ore more linter warnings / errors, all points (10) are subtracted.
+#   - For [15,25) linter warnings / errors, 6 points are subtracted.
+#   - For [7,15) linter warnings / errors, 4 points are subtracted.
+#   - For [1,7) linter warnings / errors, 2 points are subtracted.
+# Final style points are between 0 and 10. Results cannot be negative.
+#
+# Result (grade) is stored in style_grade.out file.
+# Collect summary in style_summary.out file.
+
+function grade_style()
+{
+    compiler_warn=$(< checker.out grep -v 'unused parameter' | grep -v 'unused variable' | \
+        grep -v "discards 'const'" | grep -c '[0-9]\+:[0-9]\+: warning:')
+
+    compiler_down=0
+    if test "$compiler_warn" -ge 15; then
+        compiler_down=10
+    elif test "$compiler_warn" -ge 10; then
+        compiler_down=6
+    elif test "$compiler_warn" -ge 5; then
+        compiler_down=4
+    elif test "$compiler_warn" -ge 1; then
+        compiler_down=2
+    fi
+
+    cpplint=$(< checker.out grep "Total errors found:" | rev | cut -d ' ' -f 1 | rev)
+    checkpatch_err=$(< checker.out grep 'total: [0-9]* errors' | grep -o '[0-9]* errors,' | \
+        cut -d ' ' -f 1 | paste -s -d '+' | bc)
+    checkpatch_warn=$(< checker.out grep 'total: [0-9]* errors' | grep -o '[0-9]* warnings,' | \
```

Comment:
```suggestion
    cpplint=$(< linter.out grep "Total errors found:" | rev | cut -d ' ' -f 1 | rev)
    checkpatch_err=$(< linter.out grep 'total: [0-9]* errors' | grep -o '[0-9]* errors,' | \
        cut -d ' ' -f 1 | paste -s -d '+' | bc)
    checkpatch_warn=$(< linter.out grep 'total: [0-9]* errors' | grep -o '[0-9]* warnings,' | \
```
The output for `cpplint` and `checkpatch` is in `linter.out`.

Reply:
@Alex-deVis , checkpatch configuration is only part of Docker. The script is supposed to run correctly, currently, only on Docker. We will update it to work correctly locally sometime later.

Reply:
I do not understand what this has to do with Docker, but I see you changed it as I suggested so it's fine.

Reply:
@Alex-deVis , I was referring this comment:

"""
 In the case of memory-allocator, it lacked the --no-tree flag.
 ```